src/Columns/ColumnVector.cpp

```diff
@@ -548,6 +553,110 @@ ColumnPtr ColumnVector<T>::replicate(const IColumn::Offsets & offsets) const
     return res;
 }
 
+
+#ifdef __SSE2__
+
+template <typename T>
+ColumnPtr ColumnVector<T>::replicateSSE2(const IColumn::Offsets & offsets) const
```

Comment:
@zzachimed please write high level description comment what we try to archive here.